From 8e9dbc62df228c9775fcb8a4760374d3344ccf4b Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Wed, 14 Nov 2018 13:53:38 +0800
Subject: [PATCH 5150/5242] MLK-20403: max9286: add hardware reset for max9286

commit  38f43d9e502c6fc813a187da9f79d886a349c56c from
https://source.codeaurora.org/external/imx/linux-imx.git

Add hardware reset for max9286 before initialization

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit dfc5061564b7d13d875f46bb883fe2e88a91e475)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/max9286.c |   10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/media/platform/imx8/max9286.c b/drivers/media/platform/imx8/max9286.c
index 85fc971..1b71c65 100644
--- a/drivers/media/platform/imx8/max9286.c
+++ b/drivers/media/platform/imx8/max9286.c
@@ -2470,6 +2470,14 @@ static void max9271_dump_registers(struct sensor_data *max9286_data, int index)
 }
 #endif
 
+static void max9286_hw_reset(struct sensor_data *max9286_data)
+{
+	gpio_set_value(max9286_data->pwn_gpio, 0);
+	udelay(200);
+	gpio_set_value(max9286_data->pwn_gpio, 1);
+	msleep(1);
+}
+
 static int max9286_hardware_preinit(struct sensor_data *max9286_data)
 {
 	u8 reg;
@@ -3216,6 +3224,8 @@ static int max9286_probe(struct i2c_client *client,
 	if (retval < 0)
 		return retval;
 
+	max9286_hw_reset(max9286_data);
+
 	clk_prepare_enable(max9286_data->sensor_clk);
 
 	max9286_data->i2c_client = client;
-- 
1.7.9.5

